### ISSUE TITLE
False positive with `no-unused-message-ids` when helper function parameter reassignment present

### ISSUE THREAD

**Issue by bmish** [not-an-aardvark/eslint-plugin-eslint-plugin#282]
In this example, the `messageId` being reported can come from a function parameter or from parameter reassignment. We should ignore this scenario because we don't know all the possible values the function could be called with.

Example:

```js

checkCall(node, node, 'unsafeCall');

function checkCall(
  reportingNode: TSESTree.Node,
  messageId: MessageIds,
): void {
  if (foo) {
    messageId = 'unsafeCallThis';
  }
  context.report({
    node: reportingNode,
    messageId: messageId,
  });
}
```

https://github.com/typescript-eslint/typescript-eslint/blob/b6e5413fb12c2b1943a288b514f864c339380388/packages/eslint-plugin/src/rules/no-unsafe-call.ts#L63

This is not specific to TypeScript.
